File: cpp/scrabble-score/scrabble_score.cpp
#include "scrabble_score.h"

namespace scrabble_score {

	const int score(const std::string& word)
	{
		int word_score = 0;

		for (const char& letter : word)
			word_score += letter_scores.at(toupper(letter));

		return word_score;
	}

}  // namespace scrabble_score

File: javascript/scrabble-score/scrabble-score.js
// ASSUMES: letter is a member of the upper-case English alphabet.
const getLetterScore = (letter) => {

  switch (letter) {
      case 'Q': case 'Z':
        return 10;
      case 'J': case 'X':
        return 8;
      case 'K':
        return 5;
      case 'F': case 'H': case 'V': case 'W': case 'Y':
        return 4;
      case 'B': case 'C': case 'M': case 'P':
        return 3;
      case 'D': case 'G':
        return 2;
      default:
        return 1; 
  }
}

export const score = (word) => {
  return [...word.toUpperCase()].map(getLetterScore).reduce((a, b) => a + b, 0);
};
